api/src/main/java/com/redhat/ipaas/api/IPaasEntity.java

```diff
@@ -15,9 +15,15 @@
  */
 package com.redhat.ipaas.api;
 
-public interface IPaasEntity {
+public abstract class IPaasEntity {
 
-	public String getId();
-	public void setId(String id);
+	abstract public String getId();
+	abstract public void setId(String id);
+	
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
The Ids are used a PK in the cache. We should not allow having multiple objects with the same ID(PK) in there. I think it's perfectly ok to override equals() to support this.